packages/agoric-cli/src/deploy.js

```diff
@@ -89,9 +103,60 @@ export default async function deployMain(progname, rawArgs, powers, opts) {
     1000,
   );
 
-  const retryWebsocket = async () => {
-    const accessToken = await getAccessToken(opts.hostport);
+  const accessToken = await getAccessToken(opts.hostport);
 
+  const jsonHttpCall = makeJsonHttpClient({ http });
+
+  const listConnections = async () => {
+    const { ok, connections } = await jsonHttpCall({
+      hostname: host,
+      port,
+      method: 'GET',
+      path: `/connections?accessToken=${encodeURIComponent(accessToken)}`,
+    });
+    assert(
+      ok === true,
+      `Expected JSON body "ok" property to be true for HTTP request for chain connections`,
+    );
+    return connections;
+  };
+
+  const getDefaultConnection = async () => {
+    const connections = await listConnections();
+    console.log('\nConnection options:');
+    const lookup = new Map(
+      connections.map((connection, index) => {
+        console.log(`${index + 1}.`, connection);
+        return [`${index + 1}`, connection];
+      }),
+    );
+    const { connectionNumber } = await inquirer.prompt({
+      name: 'connectionNumber',
+      message: 'Please choose a connection to publish to',
+      choices: lookup.keys(),
+    });
+    const connection = lookup.get(connectionNumber);
+    console.log({ connection });
+    return connection;
+  };
+
```

Comment:
- This is interactive
- Perhaps it should not prompt if there is 1 option
- Perhaps it should err if there are 0 options
- Perhaps interactive mode should be overridden by a command line option, if present, but describing connections requires one of various combinations of flags

Please make your preference known :-) @michaelfig @warner

Reply:
Thanks for asking!  I'd like a flag describing the desired installation target:

* `--target=local` for the local solo's own kernel (targeting spawner)
* `--target=cosmos` for the cosmos-based chain
* `--target=sim` for the sim-chain
* and an alias `--target=agoric` for the union of `cosmos` and `sim` connections

I think the default should be `--target=agoric`.

Then, choose the connection automatically if `--target` is specified and a single connection matches.    Otherwise, if stdin is a TTY, give a prompt for all connections, moving any ones matching `--target` to be first in the list.  The final fallback if stdin is not a TTY is to print an error message and exit.